File: frontend/src/types/research.ts
export interface CreateProposalRequest {
  title: string
  description: string
}

export interface ProposalItem {
  id: string
  title: string
  researcher_id: string
}

export interface CreateGrantRequest {
  title: string
  amount: number
}

export interface CreatePublicationRequest {
  title: string
  journal: string
  publication_date: string
  doi?: string
}

export interface PublicationItem {
  id: string
  title: string
  journal: string
}

File: backend/app/infrastructure/models/research.py
from beanie import Document
from pydantic import Field
from typing import Optional, List
from datetime import datetime

class ResearchProposal(Document):
    tenant_id: str
    researcher_id: str

    title: str
    description: str
    status: str = "draft"

    submitted_date: Optional[datetime] = None
    approved_date: Optional[datetime] = None

    class Settings:
        name = "research_proposals"

class Grant(Document):
    tenant_id: str
    researcher_id: str

    title: str
    amount: float
    status: str = "pending"

    created_at: datetime = Field(default_factory=datetime.utcnow)

    class Settings:
        name = "grants"

class Publication(Document):
    tenant_id: str
    researcher_id: str

    title: str
    journal: str
    publication_date: datetime
    doi: Optional[str] = None

    class Settings:
        name = "publications"
